pyproject.toml

```diff
@@ -32,7 +32,7 @@ classifiers = [
     "Operating System :: Unix",
     "Operating System :: MacOS",
 ]
-dependencies = ["numpy", "openqasm3[parser]>=1.0.0,<2.0.0"]
+dependencies = ["numpy", "matplotlib", "openqasm3[parser]>=1.0.0,<2.0.0"]
```

Comment:
let's make matplotlib an optional dependency under a `visualization` extra  instead of a core dependency e.g.

```bash
pip install 'pyqasm[visualization]'
```

See: https://packaging.python.org/en/latest/guides/writing-pyproject-toml/#dependencies-and-requirements

and then within the `.draw()` method we can catch any `ImportError` and re-raise with an exception that prompts them to install the visualization extra. For example:

https://github.com/Qiskit/rustworkx/blob/752555d2b22000784464db90187c3afba60f72b1/rustworkx/visualization/matplotlib.py#L198-L205